cmd/virt-launcher/sock-connector

```diff
@@ -1,4 +1,28 @@
 #!/bin/bash
-stty -echo
+
 SOCKET=$1
+function serial_cleanup() {
+	local pid_file=${SOCKET}.pid
+	local file_name=$(basename $SOCKET)
+
+	# if this is a serial connection, see if there is a previous
+	# connection that must be cleaned up before starting a new one.
+	if [[ $file_name =~ .*serial.* ]]; then
+		local pid=$(cat ${SOCKET}.pid 2>/dev/null)
+		local my_pid=$$
+		if [ -n "$pid" ] && [ -f "/proc/$pid/cmdline" ]; then
+			local cmdline=$(cat /proc/$pid/cmdline 2>/dev/null | tr -d '\0')
+			local my_cmdline=$(cat /proc/$my_pid/cmdline 2>/dev/null | tr -d '\0')
+			if [ "$cmdline" = "$my_cmdline" ]; then
```

Comment:
String comparisons use `==`. Are you sure this works?

Reply:
== is a synonym for = in bash.  I think some older shell's only recognize '=' for string comparisons, so I've been in the habit of only using a single '=' for a few years now. They behave the same though.

```
[dvossel@localhost ~]$ if [ "something" = "something" ]; then echo "correct"; else echo "incorrect"; fi
correct
[dvossel@localhost ~]$ if [ "something" = "somethingi" ]; then echo "incorrect"; else echo "correct"; fi
correct
[dvossel@localhost ~]$ if [ "something" == "something" ]; then echo "correct"; else echo "incorrect"; fi
correct
[dvossel@localhost ~]$ if [ "something" == "somethingi" ]; then echo "incorrect"; else echo "correct"; fi
correct
```

Reply:
I stand corrected. How'd I miss that all this time? :)